flink-runtime/src/test/java/org/apache/flink/runtime/jobmanager/Tasks.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.jobmanager;
+
+import org.apache.flink.runtime.execution.Environment;
+import org.apache.flink.runtime.io.network.api.reader.RecordReader;
+import org.apache.flink.runtime.io.network.api.writer.RecordWriter;
+import org.apache.flink.runtime.io.network.api.writer.RecordWriterBuilder;
+import org.apache.flink.runtime.jobgraph.tasks.AbstractInvokable;
+import org.apache.flink.types.IntValue;
+
+import java.io.IOException;
+
+/** Collection of {@link AbstractInvokable}s. */
+public class Tasks {
+
+    /** An {@link AbstractInvokable} that forwards all incoming elements. */
+    public static class Forwarder extends AbstractInvokable {
+
+        public Forwarder(Environment environment) {
+            super(environment);
+        }
+
+        @Override
+        public void invoke() throws Exception {
+
+            final RecordReader<IntValue> reader =
+                    new RecordReader<>(
+                            getEnvironment().getInputGate(0),
+                            IntValue.class,
+                            getEnvironment().getTaskManagerInfo().getTmpDirectories());
+
+            final RecordWriter<IntValue> writer =
+                    new RecordWriterBuilder<IntValue>().build(getEnvironment().getWriter(0));
+
+            try {
+                while (true) {
+                    final IntValue record = reader.next();
+
+                    if (record == null) {
+                        return;
+                    }
+
+                    writer.emit(record);
+                }
+
+                // writer.flushAll();
```

Comment:
uhh....the line existed in the scala version but wasn't reachable, and on the java side this actually turned into a compile error. I I commented it temporarily to see whether the tests still pass, and since they O I will remove it.

Reply:
ah true, I should have checked the code above...